core/src/main/java/org/elasticsearch/cluster/routing/allocation/command/AllocateEmptyPrimaryAllocationCommand.java

```diff
@@ -72,6 +74,18 @@ public String name() {
         return NAME;
     }
 
+    @Override
+    public String description() {
+        return String.format(Locale.ROOT, "allocate an empty primary for [%1$s][%2$s] on node [%3$s]", index, shardId, node);
+    }
+
+    @Override
+    public Optional<String> getMessage() {
```

Comment:
Question, why did you do it here and not as part of the `RoutingExplaination` returned by execute()? I'm ok if you prefer this, but just wondering to your thinking?